server/api/api/utils.py

```diff
@@ -1314,6 +1318,19 @@ async def _delete_function(
         project,
         function_name,
     )
+
+    logger.debug(
+        "Updating functions with deletion task id",
+        function_name=function_name,
+        functions_count=len(functions),
+        project=project,
+    )
+
+    # update functions with deletion task id
+    await _update_functions_with_deletion_task_ids(
+        db_session, functions, project, background_task_name
+    )
+
     if len(functions) > 0:
```

Comment:
can `len(functions)` be Zero?
If so we can skip everything, no?

Reply:
@TomerShor technically, yes, it's possible, because user can send a deletion request of the function which doesn't exist

Reply:
@TomerShor moved function deletion under the condition 👍

Reply:
Then let's move the `len` check right after getting the functions.
If there are not function - log that there is nothing to do and return